eatin-client/src/homePage/HomePage.tsx

```diff
@@ -1,5 +1,176 @@
+    const [searchFilter, setSearchFilter] = useState(false);
+    const [genreFilterVal, setGenreFilterVal] = useState("1");
+    const [difficultyFilterVal, setDifficultyFilterVal] = useState("1");
+    const [ratingFilterVal, setRatingFilterVal] = useState("1");
+    const [filterSearchVal, setfilterSearchVal] = useState("");
+    const [SearchResult, setSearchResult] = useState("");
+    const [currentShownRecipes, setCurrentShownRecipes] = useState<
+        {
+            name: string;
+            items: Item[];
+        }[]
+    >([]);
+
+    const eggs = useGetTopRatedRecipesByCategoryQuery({ variables: { category: Category.Egg } })
+        .data?.topRecipesByCategory;
+    const cakes = useGetTopRatedRecipesByCategoryQuery({ variables: { category: Category.Cake } })
+        .data?.topRecipesByCategory;
+    const japanese = useGetTopRatedRecipesByCuisineQuery({
+        variables: { cuisine: Cuisine.Japanese },
+    }).data?.topRecipesByCuisine;
+    const greek = useGetTopRatedRecipesByCuisineQuery({ variables: { cuisine: Cuisine.Greek } })
+        .data?.topRecipesByCuisine;
+
+    // This is temp
+    function getItemsArray(recipes: any) {
+        const items: Item[] = [];
+        recipes?.forEach(
+            (recipe: {
+                index: string;
+                recipe_title: string;
+                url: string;
+                record_health: string;
+                vote_count: string;
+                rating: string;
+                description: string;
+                cuisine: string;
+                course: string;
+                diet: string;
+                prep_time: string;
+                cook_time: string;
+                ingredients: string;
+                instructions: string;
+                author: string;
+                tags: string;
+                category: string;
+                image: string;
+                difficulty: string;
+            }) =>
+                items.push(
+                    new Item(
+                        recipe.index,
+                        recipe.recipe_title,
+                        recipe.url,
+                        recipe.record_health,
+                        recipe.vote_count,
+                        recipe.rating,
+                        recipe.description,
+                        recipe.cuisine,
+                        recipe.course,
+                        recipe.diet,
+                        recipe.prep_time,
+                        recipe.cook_time,
+                        recipe.ingredients,
+                        recipe.instructions,
+                        recipe.author,
+                        recipe.tags,
+                        recipe.category,
+                        recipe.image,
+                        recipe.difficulty,
+                    ),
+                ),
+        );
+        return items;
+    }
+
+    // This is temp
+    useEffect(() => {
+        const eggItems = getItemsArray(eggs);
```

Comment:
are you fixing this in your next issue? can you open an issue in our board for this
also change for `TODO` so we won't miss any comments